pkg/ccl/importccl/import_stmt.go

```diff
@@ -685,11 +690,45 @@ func importPlanHook(
 		if err != nil {
 			return err
 		}
-		return <-errCh
+		return copyToResultsChannel(ctx, errCh, resultsCh, jobResultsCh)
 	}
 	return fn, backupccl.RestoreHeader, nil, false, nil
 }
 
+// copyToResultsChannel exists to deal with the fact that the resultsCh hooked
+// up to the conn executor does not notify the plan hook when the connection
+// goes away. When that connection goes away however, the context will be
+// canceled. Because the started job needs to send on the resultsFromJobs
+// channel in order to complete, this function launches a goroutine to consume
+// from that channel in the case of context cancellation.
+func copyToResultsChannel(
+	ctx context.Context,
+	errCh <-chan error,
+	resultsCh chan<- tree.Datums,
+	resultsFromJob <-chan tree.Datums,
+) error {
+	readResultsFromJob := func() {
+		for range resultsFromJob {
+		}
+	}
+	for {
+		select {
+		case r := <-resultsFromJob:
```

Comment:
if `resultsFromJob`  is closed r is nil, right? will this just inf then? loop? should we check `errCh` first?